net-firewall/arno-iptables-firewall/arno-iptables-firewall-2.1.0.ebuild

```diff
@@ -0,0 +1,101 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit readme.gentoo-r1 systemd
+
+DESCRIPTION="Arno's iptables firewall script"
+HOMEPAGE="https://rocky.eld.leidenuniv.nl"
+
+MY_PV=$(ver_rs 3 -)
+MY_PV=${MY_PV/rc/RC}
+SRC_URI="https://github.com/${PN}/aif/archive/${MY_PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+plugins rsyslog"
+
+# sys-apps/coreutils dependency wrt
+# https://bugs.gentoo.org/show_bug.cgi?id=448716
+
+DEPEND=""
+RDEPEND="net-firewall/ipset
+	net-firewall/iptables
+	>=sys-apps/coreutils-8.21
+	sys-apps/iproute2
+	plugins? ( net-dns/bind-tools )"
+
+S="${WORKDIR}/aif-${MY_PV}"
+
+DISABLE_AUTOFORMATTING="yes"
+DOC_CONTENTS="You will need to configure /etc/${PN}/firewall.conf
+before using this package. To start the script, run:
+
+/etc/init.d/${PN} start (for OpenRC)
+systemctl start ${PN} (for systemd)
+
+If you want to start this script at boot, run:
+
+rc-update add ${PN} default (for OpenRC)
+systemctl enable ${PN} (for systemd)"
+
+src_prepare() {
+	sed -i -e 's:/usr/local/share/:/usr/libexec/:' \
+		etc/"${PN}"/firewall.conf || die "Sed failed!"
+	sed -i -e 's:/usr/local/sbin/:/usr/sbin/:' \
+		lib/systemd/system/"${PN}.service" || die "Sed failed!"
+	eapply_user
+}
+
+src_install() {
+	insinto /etc/"${PN}"
+	doins etc/"${PN}"/firewall.conf
+	doins etc/"${PN}"/custom-rules
+
+	doinitd "${FILESDIR}/${PN}"
+	systemd_dounit lib/systemd/system/"${PN}.service"
+
+	dobin bin/arno-fwfilter
+	dosbin bin/"${PN}"
+
+	insinto /usr/libexec/"${PN}"
+	doins share/"${PN}"/environment
+
+	dodoc CHANGELOG README
+	readme.gentoo_create_doc
+
+	if use plugins
+	then
+		insinto /etc/"${PN}"/plugins
+		doins etc/"${PN}"/plugins/*
+
+		insinto /usr/libexec/"${PN}"/plugins
+		doins share/"${PN}"/plugins/*.plugin
+
+		exeinto /usr/libexec/"${PN}"/plugins
+		doexe share/"${PN}"/plugins/dyndns-host-open-helper
+		doexe share/"${PN}"/plugins/traffic-accounting-helper
+		doexe share/"${PN}"/plugins/traffic-accounting-log-rotate
+		doexe share/"${PN}"/plugins/traffic-accounting-show
+
+		docinto plugins
+		dodoc share/"${PN}"/plugins/*.CHANGELOG
+	fi
+
+	if use rsyslog
+	then
+		insinto /etc/rsyslog.d
+		newins etc/rsyslog.d/"${PN}".conf 60-"${PN}".conf
+	fi
```

Comment:
What happens if I enable this USE flag, and don't have `rsyslog` installed? Your ebuild is not pulling it either. Will the conf be loaded? Will it break? Can this be installed everytime unconditionally regardless of state of having rsyslog installed or not? Would it make more sense?

Reply:
I looked for examples in existing ebuilds already in the tree about how to install an rsyslog config file properly, assuming anything I find would be an acceptable way to do it, or at least a starting point. The only example I found was from sys-cluster/glusterfs, so that's mostly why I did the exact same logic here.

> What happens if I enable this USE flag, and don't have `rsyslog` installed?

It still installs the configfile for rsyslog into `/etc/rsyslog.d`, which probably won't be picked by anything.

> Your ebuild is not pulling it either.

I initially added rsyslog as a dependency when the USE flag is enabled, but then I decided to leave it out as it doesn't feel like a "true" dependency to arno-iptables-firewall, since it would work the same regardless of rsyslog being present or not.

The iptables rules generated by the arno-iptables-firewall scripts would keep emitting their log messages as usual, and the scripts or iptables itself doesn't care about which syslog solution collects those messages to process them.

The glusterfs ebuild doesn't pull rsyslog in either, probably for similar reasons.

> Will the conf be loaded?

No, arno-iptables-firewall itself doesn't use it directly, only rsyslog would.

> Will it break?

No, I don't think arno-iptables-firewall would break, if the rsyslog config file is not present.

> Can this be installed everytime unconditionally regardless of state of having rsyslog installed or not? Would it make more sense?

It might interfere with user-supplied rsyslog configuration, so I think it's important to have a way to control installation of upstream's rsyslog config file, instead of installing it unconditionally.

Alternatively, the config file could be installed into `/usr/share/doc` as an example config, and a postinstall message could mention something like "If you want to use upstream's rsyslog config, copy this file to the proper location". IMHO the USE flag controlling the installation provides a smoother experience to the users, though.

Still, I'm happy to adapt the ebuild based on further feedback, if needed.